Mathlib/Tactic/Linter/UnusedSetOptionIn.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Tactic.Lemma
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+The `unusedSetOptionIn` linter warns against "most external" unused `set_option ... in`s.
```

Comment:
how about "... against outer-most `set option ... in`s which are not necessary"?

Reply:
Also overhauled by the new docs on the linter.